subroutines/habits/models/habits.py

```diff
@@ -0,0 +1,50 @@
+from datetime import date
+
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from recurrence import fields
+
+from subroutines.utils.models import SubRoutinesModel
+
+
+class HabitManager(models.Manager):
+    """Habit Manager with utilities to create stats"""
+
+    def get_total_active_habits_for_user(self, user):
+        """Returns the total of no paused habits for a user."""
+        return self.filter(user=user, is_paused=False).count()
+
+    def get_active_habits_for_user(self, user):
+        """Returns a list of all active habits for a user."""
+        return self.filter(user=user, is_paused=False)
+
+
+class Habit(SubRoutinesModel):
+    """
+    Habit model for subroutines app. Extend from django abstract model
+    to define extra fields needed to logic bussiness.
+    """
+
+    name = models.CharField(_("name"), max_length=50)
+    description = models.CharField(_("description"), max_length=200)
+    start_date = models.DateField(_("start date"), default=date.today)
+    end_date = models.DateField(_("end date"), blank=True, null=True)
+    is_completed = models.BooleanField(_("completed"), default=False)
+    is_paused = models.BooleanField(_("paused"), default=False)
+    is_public = models.BooleanField(_("public"), default=False)
+    recurrence = fields.RecurrenceField()
+
+    # Habit stats
+    total_instances = models.PositiveIntegerField(default=0)
+    total_instances_done = models.PositiveIntegerField(default=0)
+    completion_rate = models.DecimalField(max_digits=2, decimal_places=1, default=0)
```

Comment:
Computed fields might be hard to maintain. Maybe look for cache alternatives or computed ones.